src/H3MP/Utils/WristMenuButtons.cs

```diff
@@ -0,0 +1,280 @@
+				var button = dest.GetComponent<Button>();
+				Component.DestroyImmediate(button);			
+				button = dest.AddComponent<Button>();
+
+				button.onClick.AddListener(callbackFactory(button));
+
+				// Shift all base buttons below down
+				var btnIndex = _wristMenuBaseButtons.IndexOf(posName);
+				for (var i = btnIndex + 1; i < _wristMenuBaseButtons.Count; i++)
+				{
+					MoveButton(_wristMenuBaseButtons[i]);
+				}
+
+				return button;
+			}
+
+			void MoveButton(string posName)
+			{
+				var posTransform = canvas.Find(posName);
+				posTransform.position += BUTTON_PLACEMENT_DIFF * Vector3.down;
+			}
+		
+			// Add new buttons from top to bottom
+			yield return CreateButton("Button_1_OptionsPanel", "H3MP_Panel", "Spawn H3MP Panel", SpawnPanel);
+			yield return CreateButton("Button_3_ReloadScene", "H3MP_Privacy", PrivacyManager.PrivacyText, PrivacySelector);
+			yield return CreateButton("Button_9_BackToMainMenu", "H3MP_Disconnect", DISCONNECT_TEXT, LeaveLobby);
+		}
+
+		// H3MP_Panel onClick
+		private UnityAction SpawnPanel(Button button)
+		{
+			return () =>
+			{
+				WristMenu.Aud.PlayOneShot(WristMenu.AudClip_Engage, 1f);
+
+				FVRViveHand hand = GM.CurrentPlayerBody.RightHand.transform.Find("PointingLaser").gameObject.activeInHierarchy
+					? GM.CurrentPlayerBody.LeftHand.GetComponent<FVRViveHand>()
+					: GM.CurrentPlayerBody.RightHand.GetComponent<FVRViveHand>();
+
+				hand.RetrieveObject(GetPanel());
+				_log.LogDebug("Spawned panel");
+			};
+		}
+
+		// H3MP_Privacy onClick
+		private UnityAction PrivacySelector(Button button)
+		{
+			return () =>
+			{
+				WristMenu.Aud.PlayOneShot(WristMenu.AudClip_Engage, 1f);
+
+				PrivacyManager.CyclePrivacy();
+
+				var buttonText = button.GetComponentInChildren<Text>().text = PrivacyManager.PrivacyText;
+				_log.LogDebug("Changed party privacy");
+			};
+		}
+
+		// H3MP_Disconnect onClick
+		private UnityAction LeaveLobby(Button button)
+		{	
+			return () =>
+			{
+				WristMenu.Aud.PlayOneShot(WristMenu.AudClip_Engage, 1f);
+				if (!AskConfirmDisconnect)
+				{
+					ResetDisconnect(button);
+					AskDisonnect_Confirm(button);
+					return;
+				}
+
+				ResetDisconnect(button);
+				for (var i = 0; i < GM.CurrentSceneSettings.QuitReceivers.Count; i++)
+				{
+					GM.CurrentSceneSettings.QuitReceivers[i].BroadcastMessage("QUIT", SendMessageOptions.DontRequireReceiver);
+				}
+				if (GM.LoadingCallback.IsCompleted)
+				{
+					SteamVR_LoadLevel.Begin(SceneManager.GetActiveScene().name, false, 0.5f, 0f, 0f, 0f, 1f);
+				}
+				//TODO: make this leave session after merge with feature/netcode-refactor
+				_log.LogDebug("Left party");
+			};
+		}
+
+		public static void ResetDisconnect(Button button)
+		{
+			AskConfirmDisconnect = false;
+			button.GetComponentInChildren<Text>().text = DISCONNECT_TEXT;
+		}
+
+		private void AskDisonnect_Confirm(Button button)
+		{
+			AskConfirmDisconnect = true;
+			button.GetComponentInChildren<Text>().text = "Confirm ???";
```

Comment:
The only use of `button` is to get a component, you should just pass the component instead.